src/com/stanimiriliev/chaptertwo/heterogeneoustree/products/Egg.java

```diff
@@ -0,0 +1,15 @@
+package com.stanimiriliev.chaptertwo.heterogeneoustree.products;
+
+/**
+ * @author Stanimir Iliev
+ * @since 21.07.2017
+ */
+public class Egg extends Products {
```

Comment:
suggest, try to avoid extending classes at all. You can implement many interfaces but try to avoid extending classes. Later you will realize that, this is hard for automatic testing, but later...

This can be interface 
this morning I made an example https://gist.github.com/mlesikov/7b0b37f7d13eb67448521f0a51b8161c
it was in case you hafe troubles, but I think you've done it